jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.cpp

```diff
@@ -1448,44 +1448,65 @@ ecma_builtin_array_prototype_object_sort (ecma_value_t this_arg, /**< this argum
 
   uint32_t len = ecma_number_to_uint32 (len_number);
 
-  MEM_DEFINE_LOCAL_ARRAY (values_buffer, len, ecma_value_t);
+  uint32_t defined_prop_count = 0;
+  /* Count number of defined properties. */
+  for (ecma_property_t *property_p = ecma_get_property_list (obj_p);
+       property_p != NULL;
+       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  {
+    defined_prop_count++;
+  }
+
+  MEM_DEFINE_LOCAL_ARRAY (values_buffer, defined_prop_count, ecma_value_t);
   uint32_t copied_num = 0;
 
   /* Copy unsorted array into a native c array. */
   for (uint32_t index = 0; index < len && ecma_is_completion_value_empty (ret_value); index++)
   {
     ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
-    ECMA_TRY_CATCH (index_value, ecma_op_object_get (obj_p, index_string_p), ret_value);
+    if (ecma_op_object_get_property (obj_p, index_string_p) != NULL)
```

Comment:
The test would be executed very long, because of iteration over all values in `[0.. len)`:


```
var a = new Array ();

a[100000] = 1;
a[1000000] = 2;

a.sort ();
```

There are also another places with the issue. For example, `ecma_op_array_object_define_own_property`.
Could you, please, raise an issue about this, with references to known places with the issue, and add `FIXME` to the code part, referencing the issue?